src/components/BottomSheetModal/index.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from "react";
+import { BackHandler, StyleSheet } from "react-native";
+import {
+  BottomSheetModal as RNBottomSheetModal,
+  BottomSheetBackdrop,
+  BottomSheetView,
+  useBottomSheetDynamicSnapPoints,
+} from "@gorhom/bottom-sheet";
+
+const INITIAL_SNAP_POINTS = ["CONTENT_HEIGHT"];
+
+export function useBottomSheetModal({ openOnMount }: { openOnMount: boolean }) {
+  const initiallyOpenedRef = React.useRef(false);
+  const sheetRef = React.useRef<RNBottomSheetModal>(null);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const closeSheet = React.useCallback(() => {
+    if (sheetRef.current) {
+      sheetRef.current.dismiss();
+      setIsOpen(false);
+    }
+  }, []);
+
+  const openSheet = React.useCallback(() => {
+    if (sheetRef.current) {
+      sheetRef.current.present();
+      setIsOpen(true);
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (!initiallyOpenedRef.current && openOnMount) {
+      initiallyOpenedRef.current = true;
+      openSheet();
+    }
+  }, [openOnMount, openSheet]);
+
+  return { sheetRef, closeSheet, openSheet, isOpen };
+}
+
+function useBackPressHandler(onHardwareBackPress?: () => void | boolean) {
+  React.useEffect(() => {
+    const onBack = () => {
+      if (onHardwareBackPress) {
+        const backPress = onHardwareBackPress();
+        if (typeof backPress === "boolean") {
+          return backPress;
+        }
+      }
+
+      // We don't allow the back press to navigate/dismiss this modal by default
+      return true;
+    };
+
+    BackHandler.addEventListener("hardwareBackPress", onBack);
+
+    return () => BackHandler.removeEventListener("hardwareBackPress", onBack);
+  }, [onHardwareBackPress]);
+}
+
+interface Props extends React.PropsWithChildren<{}> {
+  onDismiss: () => void;
```

Comment:
is this `onDismiss`  callback required?

Reply:
probably doesn't need to be required no. think it was useful before because it's a specific lifecycle event from the modal api: https://gorhom.github.io/react-native-bottom-sheet/modal/props#ondismiss, but probably not needed for now